src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,177 @@
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.openOrdersList.getNext();
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping OpenProcessor Thread...");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				LOGGER.error("Error while trying to Process an Open Order with the OpenProcessor Thread", e);
+			}
+		}
+	}
+
+	/**
+	 * Method that try to get an Instance for an Open Order. This method can
+	 * generate a race condition. For example: a user can delete a Open Order
+	 * while this method is trying to get an Instance for this Order.
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
+
+				try {
+					InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+					LOGGER.info("Processing Order [" + order.getId() + "]");
+					order.handleOpenOrder();
+					OrderInstance orderInstance = instanceProvider.requestInstance(order);
+					order.setOrderInstance(orderInstance);
+
+					LOGGER.info("Removing Order [" + order.getId() + "] from Open Orders List");
+					this.openOrdersList.removeItem(order);
+
+					LOGGER.info("Updating Order State after processing [" + order.getId() + "]");
+					this.updateOrderStateAfterProcessing(order);
+
+				} catch (Exception e) {
+					LOGGER.error("Error while trying to get an Instance for Order: " + System.lineSeparator() + order,
+							e);
+					order.setOrderState(OrderState.FAILED);
+
+					LOGGER.info("Adding Order [" + order.getId() + "] to Failed Orders List");
+					this.failedOrdersList.addItem(order);
+				}
+			}
+		}
+	}
+
+	/**
+	 * After processing an Open Order, is necessary update it state.
+	 * 
+	 * @param order
+	 */
+	protected void updateOrderStateAfterProcessing(Order order) {
+		if (order.isLocal(this.localMemberId)) {
+			OrderInstance orderInstance = order.getOrderInstance();
+			String orderInstanceId = orderInstance.getId();
+
+			if (!orderInstanceId.isEmpty()) {
+				LOGGER.info("The open order [" + order.getId() + "] got an local instance with id [" + orderInstanceId
+						+ "], setting your state to SPAWNING");
+
+				order.setOrderState(OrderState.SPAWNING);
+
+				LOGGER.info("Adding Order [" + order.getId() + "] to Spawning Orders List");
+				this.spawningOrdersList.addItem(order);
+
+			} else {
+				LOGGER.error("Order Instance Id for Order [" + order.getId() + "] is Empty");
+				throw new RuntimeException("Order Instance Id for Order [" + order.getId() + "] is Empty");
+			}
+
+		} else if (order.isRemote(this.localMemberId)) {
```

Comment:
I think that it should be only a else statement

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/71e95e679ef7a149f19ffe48e1e4c50ba4fdfd0d).